src/main/java/com/wb/service/WeatherService.java

```diff
@@ -0,0 +1,66 @@
+package com.wb.service;
+
+import com.wb.payload.Weather;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.AsyncResult;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.concurrent.Future;
+
+@Service
+public class WeatherService {
+
+	private static final Logger logger = LoggerFactory.getLogger(WeatherService.class);
+	private final RestTemplate restTemplate;
+
+	@Value("${open.weather.appid}")
```

Comment:
Często spotykane w tutorialach - wole użycie `@Value` na parametrze w konstruktorze. Wtedy zmienna tez moze byc finalna. To nie jest stała więc nazwa nie powinna być z wielkich liter